**Задача 1.** Даны две  независимые выборки. 

Не соблюдается условие нормальности

x1  380,420, 290

y1 140,360,200,900

Сделайте вывод по результатам, полученным с помощью функции

Критерий Манна-Уитни:


In [9]:
import numpy as np
from scipy import stats


x = np.array([380, 420, 290])
y = np.array([140, 360, 200, 900])
stats.mannwhitneyu(x, y)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

Вывод: Смотрим в пользу нулевой гипотезы.

**Задача 2.** Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150,  130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

Критерий Фридмана:

alpha = 0.05

In [8]:
measuring_1 = ([150, 160, 165, 145, 155])
measuring_2 = ([140, 155, 150, 130, 135])
measuring_3 = ([130, 130, 120, 130, 125])

stats.friedmanchisquare(measuring_1, measuring_2, measuring_3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

Вывод: Статистические различия есть.

**Задача 3.** Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

Критерий Уилкоксона:

In [7]:
stats.wilcoxon(measuring_1, measuring_2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

Вывод: Статистических различий нет (pvalue = 0.0625 > alpha = 0.05)

**Задача 4.** Даны 3 группы  учеников плавания.

В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

Проверить, есть ли различия между группами

Критерий Крускала – Уоллиса:

In [6]:
gr_1 = ([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
gr_2 = ([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
gr_3 = ([57, 67, 49, 48, 47, 55, 66, 51, 54])
stats.kruskal(gr_1, gr_2, gr_3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

Вывод: Статистических различий нет (pvalue = 0.065 > alpha = 0.05)

**Задача 5.** Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [ ]:

x = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
mu = 2.5

def t_critery(x, mu):
    t = (np.mean(x) - mu) / np.std(x, ddof=1) * np.sqrt(len(x))
    t_table = stats.t.ppf(0.975, (len(x) - 1))
    return "Вывод: Детали из выборки принадлежат партии" if t_table > t else "Вывод:Детали из выборки не из этой партии"


t_critery(x, mu)

Вывод:   Детали из выборки принадлежат партии